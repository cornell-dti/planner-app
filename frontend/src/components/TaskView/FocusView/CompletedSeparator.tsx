import React, { ReactElement } from 'react';
import styles from './CompletedSeparator.module.scss';
import SamwiseIcon from '../../UI/SamwiseIcon';

type Props = {
  readonly count: number;
  readonly doesShowCompletedTasks: boolean;
  readonly onDoesShowCompletedTasksChange: () => void;
};

const getIconClassName = (notInverted: boolean): string => (
  notInverted ? styles.Icon : `${styles.Icon} ${styles.Inverted}`
);

export default (
  { count, doesShowCompletedTasks, onDoesShowCompletedTasksChange }: Props,
): ReactElement => (
  <div className={styles.Separator}>
    <span className={styles.Text}>{`Completed: (${count})`}</span>
    <SamwiseIcon
      iconName="dropdown"
      className={getIconClassName(doesShowCompletedTasks)}
      onClick={onDoesShowCompletedTasksChange}
    />
    <div className={styles.Line} />
  </div>
);
