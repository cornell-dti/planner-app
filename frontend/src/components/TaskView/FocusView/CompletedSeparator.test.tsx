import React from 'react';
import renderer from 'react-test-renderer';
import CompletedSeparator from './CompletedSeparator';

it('CompletedSeparator when showing completed tasks matches snapshot', () => {
  const tree = renderer.create(
    <CompletedSeparator
      count={10}
      doesShowCompletedTasks
      onDoesShowCompletedTasksChange={() => { }}
    />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('CompletedSeparator when not showing completed tasks matches snapshot', () => {
  const tree = renderer.create(
    <CompletedSeparator
      count={5}
      doesShowCompletedTasks={false}
      onDoesShowCompletedTasksChange={() => { }}
    />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
