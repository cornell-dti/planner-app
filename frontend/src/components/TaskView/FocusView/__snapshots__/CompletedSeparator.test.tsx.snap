// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompletedSeparator when not showing completed tasks matches snapshot 1`] = `
<div
  className="Separator"
>
  <span
    className="Text"
  >
    Completed: (5)
  </span>
  <span
    onClick={[Function]}
    onKeyUp={[Function]}
    role="button"
    tabIndex={0}
    title="More"
  >
    <img
      alt="More"
      className="SamwiseIconDefaultStyle Icon Inverted"
      src="/svgs/v.svg"
    />
  </span>
  <div
    className="Line"
  />
</div>
`;

exports[`CompletedSeparator when showing completed tasks matches snapshot 1`] = `
<div
  className="Separator"
>
  <span
    className="Text"
  >
    Completed: (10)
  </span>
  <span
    onClick={[Function]}
    onKeyUp={[Function]}
    role="button"
    tabIndex={0}
    title="More"
  >
    <img
      alt="More"
      className="SamwiseIconDefaultStyle Icon"
      src="/svgs/v.svg"
    />
  </span>
  <div
    className="Line"
  />
</div>
`;
